lms/tasks/email_digests.py

```diff
@@ -95,24 +86,20 @@ def send_instructor_email_digest_tasks(self, *, batch_size, h_userids=None):
                             "updated_before": updated_before,
                         },
                     )
-                except Exception as exc:  # pylint:disable=broad-exception-caught
-                    LOG.exception(exc)
-                    report_exception(exc)
-                    failed_h_userids.extend(batch)
-
-            if failed_h_userids:
-                self.retry(
-                    kwargs={
-                        **self.request.kwargs,
-                        "h_userids": failed_h_userids,
-                    },
-                    countdown=_retry_countdown(self.request),
+                except Exception as err:  # pylint:disable=broad-exception-caught
+                    failed = True
+                    LOG.exception(err)
+                    report_exception(err)
+
+            if failed:
+                raise RuntimeError(
+                    "One or more send_instructor_email_digests() batches failed to schedule"
```

Comment:
If we did catch any exceptions in the `for` loop above we need to remember that and raise here so that Celery will retry the task so that we'll try again to schedule the failed batches. We will also re-schedule the batches that succeeded which will result in unnecessary processing but due to the `task_done` table it shouldn't send any duplicate emails